client/matching/client.go

```diff
@@ -158,7 +158,7 @@ func (c *clientImpl) QueryWorkflow(ctx context.Context, request *matchingservice
 	if err != nil {
 		return nil, err
 	}
-	ctx, cancel := c.createContext(ctx)
+	ctx, cancel := c.createLongPollContext(ctx)
```

Comment:
This is not long poll call, right? You just need a longer timeout. I would suggest to create another method `createContextWithTimeout` and pass `QueryTimeout` const to it.

Reply:
the caller will be waiting on matching service for the result, similar to long poll.
so yes longer timeout, and similar to long poll....

i can create a dedicated function if you want